### PR TITLE
Add dotenv 

### PR DESCRIPTION
**Pull request by rileyanderson**
Adds dotenv for env management locally.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -37,3 +37,6 @@ test/reports
 # Ignore Tailwind generated files
 /app/assets/builds/*
 !/app/assets/builds/.keep
+
+#Ignore local env file
```
Comment (ericboehs) [inline .gitignore:41]:
```suggestion
# Ignore local env file
```
Eww

File: lib/dotenv.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    string = string[1...-1] if string.end_with? "'", '"'
+    string
+  end
+
+  def replace_newline_character(string)
+    string.gsub '\n', "\n"
+  end
+end
+
+Dotenv.new.load %w[.env.local] if ENV['RAILS_ENV'] == 'development' || ENV['RAILS_ENV'] == 'test'
```
Comment (ericboehs) [inline lib/dotenv.rb:48]: You'll also need `.env` in this list.

File: .env [hunk 3]
```diff
@@ -0,0 +1 @@
+GITHUB_ACCESS_TOKEN=xxx
```
Comment (ericboehs) [inline .env:1]: Did you mean to name this .env.example?

File: config/initializers/octokit.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+# frozen_string_literal: true
+
+Octokit.configure do |config|
+  config.access_token = ENV['GITHUB_ACCESS_TOKEN']
```
Comment (ericboehs) [inline config/initializers/octokit.rb:4]:
```suggestion
  config.access_token = ENV.fetch 'GITHUB_ACCESS_TOKEN'
```

Should we raise if this isn't set so we know why github integration isn't working?

File: config/initializers/dotenv.rb [hunk 5]
```diff
@@ -0,0 +1,3 @@
+# frozen_string_literal: true
+
+require 'dotenv'
```
Comment (ericboehs) [inline config/initializers/dotenv.rb:3]: You could also include this at the top of application.rb (either way).

**Comment by ericboehs** [comment]
This PR doesn't implement any functionality. Should we wait until these gems are actually used before merging to master?

File: .env (hunk 3 above)
Comment (rileyanderson) [reply .env:1]: indeed

File: config/initializers/dotenv.rb (hunk 5 above)
Comment (rileyanderson) [reply config/initializers/dotenv.rb:3]: Umm i kept getting an error when putting it there but i may need to change some of the autoloading settings

**Comment by rileyanderson** [comment]
> This PR doesn't implement any functionality. Should we wait until these gems are actually used before merging to master?

@ericboehs True. I mainly just wanted to get the dotenv stuff in. I can break it out

**Review by thilton-oddball — APPROVED** [review]

**Pull request closed without merging**